### PR TITLE
#164798909 Implement social login functionality

### PR DESCRIPTION
**Pull request by TheDrizzyWay**
#### Description
Implement a functionality which enables users sign in to the application via facebook, twitter, or google.

#### Type of change
- New feature

#### Checklist:
- [x]  Write a test for social authentication.
- [x] Generate application keys on facebook, twitter and google
- [x]  Set up middleware which interfaces with social sites
- [x] Create controller function which redirects the user.
- [x] Add API documentation using swagger

#### PIVOTAL BOARD
[#164798909](https://www.pivotaltracker.com/story/show/164798909)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "pretest": "cross-env NODE_ENV=test && cross-env NODE_ENV=test npm run migrate:undo && cross-env NODE_ENV=test npm run migrate",
+    "pretest": "npm run migrate:undo && npm run migrate && npm run seed",
```
Comment (paradisekelechi) [inline package.json]: Why did you remove cross-env here?

Comment (TheDrizzyWay) [reply package.json]: I must have removed it while resolving merge conflicts. Will fix that right away

Comment (TheDrizzyWay) [reply package.json]: Now fixed.

**Comment by chukwudidu** [comment]
Looks good.

**Review by davidokonji — COMMENTED** [review]
Good work.
